lazy_dataset/core.py

```diff
@@ -2818,95 +2835,51 @@ def __del__(self):
-                        f'dir "{self.cache.cache.directory}"! (total='
-                        f'{humanfriendly.format_size(diskusage.total, binary=True)}'
-                        f', free='
-                        f'{humanfriendly.format_size(diskusage.free, binary=True)}'
-                        f')'
-                    )
-                    if diskusage.free < 1*1024**3:
-                        # Crash if less than 1GB left. It's better to crash
-                        # this process than to crash the whole machine
-                        raise RuntimeError(
-                            f'Not enough space on device! The device that the '
-                            f'cache directory is located on has less than 1GB '
-                            f'space left. You probably want to delete some '
-                            f'files before crashing the machine.'
-                        )
-
-                self.cache.cache[item] = self.input_dataset[item]
-            return self.cache.cache[item]
+            self._cache = _DiskCacheWrapper(cache_dir, reuse, clear)
         else:
-            # Support for slices etc.
-            return super().__getitem__(item)
+            self._cache = _cache
 
-    def copy(self, freeze: bool = False) -> 'Dataset':
-        if not freeze:
+    def check(self):
+        import shutil
+        import humanfriendly
+        diskusage = shutil.disk_usage(self._cache.cache.directory)
+        if diskusage.free < 5 * 1024 ** 3:
             import warnings
             warnings.warn(
-                'Copying a CacheDataset preserves the cache, i.e., the '
-                'already cached part of the dataset will be frozen even if '
-                'freeze=False!'
+                f'There is not much space left in the specified cache '
+                f'dir "{self._cache.cache.directory}"! (total='
+                f'{humanfriendly.format_size(diskusage.total, binary=True)}'
+                f', free='
+                f'{humanfriendly.format_size(diskusage.free, binary=True)}'
+                f')'
             )
-        # We have to share the cache here because otherwise a new cache would
-        # be initialized at every copy and copy is called by prefetch before
-        # iterating over the dataset
-        copy = self.__class__(
-            self.input_dataset.copy(freeze),
-            _cache=self.cache
-        )
-
-        return copy
+            if diskusage.free < 1 * 1024 ** 3:
+                # Crash if less than 1GB left. It's better to crash
+                # this process than to crash the whole machine
+                raise RuntimeError(
+                    f'Not enough space on device! The device that the '
+                    f'cache directory is located on has less than 1GB '
+                    f'space left. You probably want to delete some '
+                    f'files before crashing the machine.'
```

Comment:
Can you add `self._cache.cache.directory` to the error msg?

Reply:
Done